plugins/modules/idrac_firmware_info.py

```diff
@@ -104,34 +106,60 @@
   }
 '''
 
-
 import json
 from ansible_collections.dellemc.openmanage.plugins.module_utils.dellemc_idrac import iDRACConnection, idrac_auth_params
+from ansible_collections.dellemc.openmanage.plugins.module_utils.idrac_redfish import iDRACRedfishAPI
+from ansible_collections.dellemc.openmanage.plugins.module_utils.utils import GET_IDRAC_FIRMWARE_DETAILS_URI_10, GET_IDRAC_FIRMWARE_URI_10, remove_key
 from ansible.module_utils.basic import AnsibleModule
 from ansible.module_utils.urls import ConnectionError, SSLValidationError
-from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+from urllib.error import URLError, HTTPError
+
+ERR_STATUS = 404
+
+
+def get_from_wsman(module):
+    with iDRACConnection(module.params) as idrac:
+        firmware_details = idrac.update_mgr.InstalledFirmware
+        return firmware_details
+
+
+def get_idrac_firmware_info(idrac, module):
+    try:
+        response = idrac.invoke_request(method='GET', uri=GET_IDRAC_FIRMWARE_URI_10)
```

Comment:
what if response status code is not 200? handle that scenario.

Reply:
handled